packages/apollo-gateway/src/index.ts

```diff
@@ -73,6 +85,38 @@ function isManagedConfig(
   return !isRemoteConfig(config) && !isLocalConfig(config);
 }
 
+type DidResolveQueryPlanCallback = ({
+  queryPlan,
+  serviceMap,
+  operationContext,
+}: {
+  queryPlan: QueryPlan;
+  serviceMap: ServiceMap;
+  operationContext: OperationContext;
+}) => void;
```

Comment:
why void? not saying that this is wrong but i wonder if it might be better to have some type instead no type?

Reply:
I like the idea that we can codify with the type system that the callback is supposed to have no side effect on state and the response will be ignored. Maybe we want to go a step further and wrap the arguments in ReadOnly?